core/src/main/java/com/datastax/oss/driver/internal/core/metadata/InitialNodeListRefresh.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.metadata;
+
+import com.datastax.oss.driver.api.core.metadata.EndPoint;
+import com.datastax.oss.driver.api.core.metadata.Node;
+import com.datastax.oss.driver.internal.core.context.InternalDriverContext;
+import com.datastax.oss.driver.internal.core.metadata.token.DefaultTokenMap;
+import com.datastax.oss.driver.internal.core.metadata.token.TokenFactory;
+import com.datastax.oss.driver.internal.core.metadata.token.TokenFactoryRegistry;
+import com.datastax.oss.driver.shaded.guava.common.annotations.VisibleForTesting;
+import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableList;
+import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableMap;
+import java.util.Set;
+import java.util.UUID;
+import net.jcip.annotations.ThreadSafe;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The first node list refresh: contact points are not in the metadata yet, we need to copy them
+ * over.
+ */
+@ThreadSafe
+class InitialNodeListRefresh extends NodesRefresh {
+
+  private static final Logger LOG = LoggerFactory.getLogger(InitialNodeListRefresh.class);
+
+  @VisibleForTesting final Iterable<NodeInfo> nodeInfos;
+  @VisibleForTesting final Set<Node> contactPoints;
```

Comment:
shouldn't it be `Set<DefaultNode>`? I see that all additions and retrievals with that `Set` are casting to that `DefaultNode` implementation

Reply:
Yes, in this particular case the set comes from `MetadataManager.getContactPoints`, which is purely internal, so I can make that change to avoid the casts.